knox-token-management-ui/token-management/app/token.management.component.html

```diff
@@ -20,90 +20,116 @@
                     <button *ngIf="!knoxToken.metadata.enabled && !isTokenExpired(knoxToken.expirationLong)" (click)="enableToken(knoxToken.tokenId);">Enable</button>
                     <button (click)="revokeToken(knoxToken.tokenId);">Revoke</button>
-                </td>
-            </tr>
-            </tbody>
-		    <tfoot>
-		    <tr>
-		        <td colspan="6">
-		            <mfBootstrapPaginator [rowsOnPageSet]="[5,10,15]"></mfBootstrapPaginator>
-		        </td>
-		    </tr>
-		    </tfoot>
-        </table>
+                </mat-cell>
+            </ng-container>
+
+            <mat-header-row *matHeaderRowDef="displayedColumns"></mat-header-row>
+            <mat-row *matRowDef="let row; columns: displayedColumns;"></mat-row>
+
+        </mat-table>
+        <mat-paginator #ownPaginator [pageSizeOptions]="[5, 10, 25, 100]" [showFirstLastButtons]="true"></mat-paginator>
     </div>
 
+
     <!-- 'doAs' Knox Tokens (tokens created by the current user on behalf on another user -->
 
-    <div class="table-responsive" style="width:100%; overflow: auto; overflow-y: scroll; padding: 10px 0px 0px 0px;" *ngIf="isImpersonationEnabled()">
-        <label>Impersonation Knox Tokens</label>
-        <table class="table table-hover" [mfData]="doAsKnoxTokens" #doAsTokens="mfDataTable" [mfRowsOnPage]="10">
-            <thead>
-            <tr>
-                <th>Token ID</th>
-                <th>Issued</th>
-                <th>Expires</th>
-                <th>Comment</th>
-                <th>Additional Metadata</th>
-                <th>Impersonated User<th>
-            </tr>
-            </thead>
-            <tbody>
-            <tr *ngFor="let doAsKnoxtoken of doAsTokens.data">
-                <td>{{doAsKnoxtoken.tokenId}}</td>
-                <td>{{formatDateTime(doAsKnoxtoken.issueTimeLong)}}</td>
-                <td *ngIf="!isTokenExpired(doAsKnoxtoken.expirationLong)" style="color: green">{{formatDateTime(doAsKnoxtoken.expirationLong)}}</td>
-                <td *ngIf="isTokenExpired(doAsKnoxtoken.expirationLong)" style="color: red">{{formatDateTime(doAsKnoxtoken.expirationLong)}}</td>
-                <td>{{doAsKnoxtoken.metadata.comment}}</td>
-                <td>
-                  <ul>
-                    <li *ngFor="let metadata of getCustomMetadataArray(doAsKnoxtoken)">
-                      {{metadata[0]}} = {{metadata[1]}}
-                    </li>
-                  </ul>
-                </td>
-                <td>{{doAsKnoxtoken.metadata.userName}}</td>
-            </tr>
-            </tbody>
-		    <tfoot>
-		    <tr>
-		        <td colspan="6">
-		            <mfBootstrapPaginator [rowsOnPageSet]="[5,10,15]"></mfBootstrapPaginator>
-		        </td>
-		    </tr>
-		    </tfoot>
-        </table>
+    <div class="table-responsive" style="width:100%; overflow: auto; overflow-y: scroll; padding: 10px 0px 0px 0px;">
```

Comment:
Why are tokens acquired with impersonation presented in a distinct table from "normal" tokens? Couldn't it be a single table? The relationship between the tables isn't entirely clear to me, and you can only act on rows in the My Knox Tokens table.

Reply:
In theory, they could be, but I made this decision when token impersonation was introduced (so this is not a new thing here). This is easier to understand and the UX is better this way (IMO, at least)
The relationship is straightforward:
- in the first table we list all tokens generated by the logged-in user for himself/herself: `userName=logged-in user`
- in the second one we list all tokens generated by the logged-in user on behalf of other users: `username=otherUser; createdBy=logged-in user`

The same is described [here](https://knox.apache.org/books/knox-2-0-0/user-guide.html#Token+Management).

Reply:
I recognize that this is not directly related to this PR.
I think my confusion may be due to the inability to disable/revoke the tokens in the second table, but that is a separate topic.